src/libraries/System.Text.Json/Common/JsonSimpleNamingPolicy.cs

```diff
@@ -0,0 +1,147 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using System.Buffers;
+using System.Globalization;
+
+namespace System.Text.Json
+{
+    internal sealed class JsonSimpleNamingPolicy : JsonNamingPolicy
+    {
+        private readonly bool _lowercase;
+        private readonly char _boundary;
+
+        internal JsonSimpleNamingPolicy(bool lowercase, char boundary) =>
+            (_lowercase, _boundary) = (lowercase, boundary);
+
+        public override string ConvertName(string name)
+        {
+            int bufferLength = name.Length * 2;
+            char[]? buffer = bufferLength > 512
+                ? ArrayPool<char>.Shared.Rent(bufferLength)
+                : null;
+
+            int resultLength = 0;
+            Span<char> result = buffer is null
+                ? stackalloc char[512]
```

Comment:
Given the rather not super common usage of these classes I'd generally prefer simplicity in the code (i.e. use StringBuilder) - the stackallocs and shared buffers are much more error prone and given we later cache the names in JsonPropertyInfo (both UTF-8 and UTF-16 versions) I don't see super much value in speeding up initialization process (these will be called exactly once per JsonSerializerOptions per member) given how much complexity this adds vs benefit I'll vote to simplify this code.

Reply:
cc: @eiriktsarpalis @layomia thoughts?

Reply:
`StringBuilder` requires allocations, at least two. There's a shared thread static instance used for formatting and some other cases in the runtime, but it cannot be accessed from the JSON library. So I would like to use fixes I already made for my library here, it's exactly what Nino proposed.

Reply:
note that it's not two allocations per serialization, it's two allocations per app runtime (assuming you use same options instance). It doesn't seem too bad to be honest, it will have negligible impact on the overall performance

Reply:
How can it be per runtime when a single converter can be used by multiple threads and there's no shared builder?

Reply:
Yeah, but how many properties are there? Now multiply that by number of allocations required or by time it takes to measure the initial slowdown which the team tries to improve in the whole .NET.

Reply:
you're not serializing entire framework, just couple of types in typical scenarios. For scenarios where you care about initialization time presumably you should use source-gen

Reply:
Couple of types with some number of fields fields, right. Just keep in mind that not anyone follow guidelines, recommendations and use source generators. There a plenty of services made low-quality just to work and then patched to include new logic and perf issue fixing in the backlog (:

I think there can be a better solution than the string builder you mentioned and mine solution. There's an internal builder included as source in some libraries, but not everywhere to avoid bloating.

Reply:
A `JsonNamingPolicy` can be used in [`DictionaryKeyPolicy`](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializeroptions.dictionarykeypolicy?view=net-6.0#system-text-json-jsonserializeroptions-dictionarykeypolicy), so it might get invoked multiple times [at serialization time](https://github.com/dotnet/runtime/blob/f7eb3c4706fb2422c702ff1ab4fdc7e1d1687954/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/StringConverter.cs#L31-L44). For that reason, I think it makes sense to try to curtail allocations.